pkg/runner/chroot_runner.go

```diff
@@ -0,0 +1,73 @@
+package runner
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/buildbarn/bb-remote-execution/pkg/proto/runner"
+	"github.com/buildbarn/bb-storage/pkg/util"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type chrootRunner struct {
+	base               Runner
+	buildDirectoryPath string
+}
+
+// NewChrootRunner modifies the arguments of the incoming RunRequest so
+// that the first argument is an absolute path resolved using the path
+// in the request using the input root as the chroot
+func NewChrootRunner(base Runner, buildDirectoryPath string) Runner {
```

Comment:
Maybe it would be better to call this Argv0ResolvingRunner, or ExecutablePathResolvingRunner?

Could it be made to work for non-chrooted requests as well? That way we could simply enable it unconditionally, thereby getting consistent behaviour between chroot and !chroot.